filelog.go

```diff
@@ -251,9 +275,63 @@ func NewFileLogWriter(fname string, rotate bool) *FileLogWriter {
 		}
 	}()
 
+	// Background tasks goroutine
+	w.wg.Add(1)
+	go func() {
+		defer w.wg.Done()
+
+		for filename := range w.backgroundTasks {
+			if w.filesToKeep > 0 {
+				dir := filepath.Dir(filename)
+				err := w.archiveFiles(dir)
+				if err != nil {
+					fmt.Fprintf(w.errorWriter, "FileLogWriter(%q): Couldn't archive files: %s\n", filename, err)
+				}
+			}
+		}
+	}()
+
 	return w
 }
 
+func (w *FileLogWriter) archiveFiles(dir string) error {
+	filenames := make([]string, 0)
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
```

Comment:
I definitely rely on the `filepath.Walk` sorting behavior. Initially I had some other reasons that I chose `filepath.Walk`, but the filter+sort strategy is probably better.